File: functions/ffmpeg/variants/go/3_weight_3.go
package main

func weight(i, blen, offset int32) int32 {
    if i < 2*offset {
        if offset == 1 {
            if i > 0 {
                return 5
            }
            return 3
        }
        return (6*i + offset - 1)/(2*offset - 1) + 1
    }
    if i > blen - 1 - 2*offset {
        if offset == 1 {
            if blen-1-i > 0 {
                return 5
            }
            return 3
        }
        return (6*(blen-1-i) + offset - 1)/(2*offset - 1) + 1
    }
    return 8
}


func main(){
    weight(0,0,0)
}

File: functions/ffmpeg/variants/c/3_weight_3.c
#include <stdint.h>

static int weight(int i, int blen, int offset)
{
    if (offset == 1) {
        return (i < 2 * offset) ? ((i) ? 5 : 3) : ((i > blen - 1 - 2 * offset) ? ((blen - 1 - i) ? 5 : 3) : 8);
    } else {
        return (i < 2 * offset) ? (1 + (6 * i + offset - 1) / (2 * offset - 1)) : ((i > blen - 1 - 2 * offset) ? (1 + (6 * (blen - 1 - i) + offset - 1) / (2 * offset - 1)) : 8);
    }
}


int main(){
    weight(0,0,0);
}
